lib/nano.js

```diff
@@ -168,29 +163,48 @@ module.exports = exports = function dbScope (cfg) {
     delete responseHeaders.server
     delete responseHeaders['content-length']
 
-    /* if (opts.dontParse) {
-      parsed = body
-    } else {
-      try { parsed = JSON.parse(body) } catch (err) { parsed = body }
-    } */
-
     if (statusCode >= 200 && statusCode < 400) {
-      log({ err: null, body, headers: responseHeaders })
-      if (resolve) {
-        resolve(body)
+      // collect response
+      const ct = response.headers.get('content-type')
+      let retval = ''
+      if (ct === 'application/json') {
+        try {
+          retval = await response.json()
+        } catch {
+          // do nothing
```

Comment:
It would. I'll fix

Reply:
I've just realised why this is there. When doing `HEAD /db/id` there is no body so response.json throws an error.

Reply:
ah, a comment would be nice then :)

Reply:
and re-throw if method != head?

Reply:
Yeah. I tried that but some tests fail. The tests (and presumably the users) are expecting an rejected promise not a thrown exception. This would apply to 500 responses (where we are expecting JSON back but get nothing)

Reply:
Added a comment to this section of the code, for future generations.